src/main/java/org/terasology/fluid/system/FluidAuthoritySystem.java

```diff
@@ -54,31 +61,30 @@
     @ReceiveEvent
     public void fillFluidContainerItem(ActivateEvent event, EntityRef item, FluidContainerItemComponent fluidContainer,
                                        ItemComponent itemComponent) {
-//        if (fluidContainer.fluidType == null || fluidContainer.volume < fluidContainer.maxVolume) {
-//            Vector3f location = event.getInstigatorLocation();
-//            Vector3f direction = new Vector3f(event.getDirection());
-//            direction.normalize();
-//            for (int i = 0; i < 3; i++) {
-//                location.add(direction);
-//                LiquidData liquid = worldProvider.getLiquid(new Vector3i(location, 0.5f));
-//                if (liquid != null && liquid.getType() != null && liquid.getDepth() > 0) {
-//                    EntityRef owner = item.getOwner();
-//                    final EntityRef removedItem = inventoryManager.removeItem(owner, event.getInstigator(), item, false, 1);
-//                    if (removedItem != null) {
-//                        String fluidType = fluidRegistry.getFluidType(liquid.getType());
-//
-//                        // Set the contents of this fluid container and fill it up to max capacity.
-//                        FluidUtils.setFluidForContainerItem(removedItem, fluidType,
-//                                removedItem.getComponent(FluidContainerItemComponent.class).maxVolume);
-//
-//                        if (!inventoryManager.giveItem(owner, event.getInstigator(), removedItem)) {
-//                            removedItem.destroy();
-//                        }
-//                    }
-//                    event.consume();
-//                    return;
-//                }
-//            }
-//        }
+        if (fluidContainer.fluidType == null || fluidContainer.volume < fluidContainer.maxVolume) {
+            Vector3f location = event.getInstigatorLocation();
+            Vector3f direction = new Vector3f(event.getDirection());
+            direction.normalize();
```

Comment:
Could these two lines be merged? Or is it only JOML that allows for it?

```suggestion
            Vector3f direction = new Vector3f(event.getDirection()).normalize();
```